pika/adapters/select_connection.py

```diff
@@ -188,23 +135,54 @@ class SelectPoller(object):
     to override the update_handler and start methods for additional types.
 
     """
-    TIMEOUT = 0.01
+    # Drop out of the poll loop every POLL_TIMEOUT secs as a worst case, this
+    # is only a backstop value. We will run timeouts when they are scheduled.
+    POLL_TIMEOUT = 5
+    # if the poller uses MS specify 1000
+    POLL_TIMEOUT_MULT = 1
 
-    def __init__(self, fileno, handler, events, state_manager):
+    def __init__(self):
         """Create an instance of the SelectPoller
 
-        :param int fileno: The file descriptor to check events for
-        :param method handler: What is called when an event happens
-        :param int events: The events to look for
-        :param method state_manager: The method to manage state
-
         """
-        self.fileno = fileno
-        self.events = events
-        self.open = True
-        self._handler = handler
-        self._timeouts = []
-        self._manage_event_state = state_manager
+        self._fd_handlers = dict()
+        self._fd_events = {READ: set(), WRITE: set(), ERROR: set()}
+        self._stopping = False
+        self._timeouts = {}
+        self._next_timeout = None
+        self._processing_fd_event_map = {}
+        self._r_interrupt, self._w_interrupt = self.get_interrupt_pair()
```

Comment:
No, the socket should be garbage collected by python.

Also remember we're talking about one pair ioloop, not per connection and
usually you'd only ever want one ioloop for the lifetime of the process.
On 4 May 2015 22:44, "vitaly-krugl" <notifications@github.com> wrote:

> In pika/adapters/select_connection.py
> <https://github.com/pika/pika/pull/555#discussion_r29627028>:
>
> >          """
> > -        self.fileno = fileno
> > -        self.events = events
> > -        self.open = True
> > -        self._handler = handler
> > -        self._timeouts = []
> > -        self._manage_event_state = state_manager
> > +        self._fd_handlers = dict()
> > +        self._fd_events = {READ: set(), WRITE: set(), ERROR: set()}
> > +        self._stopping = False
> > +        self._timeouts = {}
> > +        self._next_timeout = None
> > +        self._processing_fd_event_map = {}
> > +        self._r_interrupt, self._w_interrupt = self.get_interrupt_pair()
>
> @wjps <https://github.com/wjps>, this change may be leaking UDP sockets,
> as there is no code that explicitly closes these two sockets self._r_interrupt,
> self._w_interrupt
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pika/pika/pull/555/files#r29627028>.
>